public/hooks/use_chat_state.tsx

```diff
@@ -122,9 +147,31 @@ const chatStateReducer: React.Reducer<ChatState, ChatStateAction> = (state, acti
           action.payload.interactions,
           'interaction_id'
         );
-        draft.llmResponding = false;
         draft.llmError = undefined;
         break;
+
+      case 'appendMessage':
+        const updatingMessage = state.messages.find(
+          (message) => message.messageId === action.payload.messageId
+        );
+        if (updatingMessage) {
+          const patchMessages: IMessage[] = [
+            {
+              ...updatingMessage,
+              content: updatingMessage.content + action.payload.content,
+            },
+          ];
+          draft.messages = addPatchInArray(state.messages, patchMessages, 'messageId');
+        }
+        break;
+
```

Comment:
How about declare `apepndMessage` like this:
```
{
      type: 'appendMessage';
      payload: {
        content: string;
        messageId: string;
        suggestedActions?: ISuggestedAction[];
      };
    };
```
and modify there to be something like:
```
          const patchMessages: IMessage[] = [
            {
              ...updatingMessage,
              content: updatingMessage.content + action.payload.content,
              ...(action.payload.suggestedActions && {
                suggestedActions: [
                  ...((updatingMessage as IOutput).suggestedActions || []),
                  ...action.payload.suggestedActions,
                ],
              }),
            },
          ];
```
so we can also handle the appending of suggested actions.

Reply:
I am thinking we should add a new action like
```
{
  event: 'updateOutputMessage',
  data: {
    messageId: string;
    payload: Partial<IOutput>
  }
}
```
to keep appendMessage clean.